pytest_plugins/xdist.py

```diff
@@ -4,24 +4,27 @@
 
 import pytest
 from broker.broker import VMBroker
+from xdist import get_xdist_worker_id
 
 import robottelo
+from pytest_fixtures.broker import satellite_factory
 from robottelo.config import settings
 
 logger = logging.getLogger('robottelo')
 
 
-@pytest.fixture(scope="session", autouse=True)
-def align_to_satellite(worker_id, satellite_factory):
-    """Attempt to align a Satellite to the current xdist worker"""
-    cache_proxy = robottelo.config.settings_proxy._cache
+@pytest.hookimpl(tryfirst=True)
+def pytest_sessionstart(session):
+    """Attempt to align a Satellite to the current xdist worker
+
+    Sets settings.server.hostname in the settings cache, since that's what the framework resolves
+    """
+    cache_proxy = robottelo.config.settings_proxy._cache  # noqa
     # clear any hostname that may have been previously set
     cache_proxy['server.hostname'] = on_demand_sat = None
 
-    if worker_id in ['master', 'local']:
-        worker_pos = 0
-    else:
-        worker_pos = int(worker_id.replace('gw', ''))
+    worker_id = get_xdist_worker_id(session)
+    worker_pos = 0 if worker_id == 'master' else int(worker_id.replace('gw', ''))
```

Comment:
```suggestion
    worker_pos = 0 if worker_id in ['master', 'local'] else int(worker_id.replace('gw', ''))
```

Reply:
Why check for local? `xdist.get_xdist_worker_id` doesn't return `'local'` AFAIK.

https://github.com/pytest-dev/pytest-xdist/blob/master/src/xdist/plugin.py#L248

Reply:
unfortunately, there are cases where local does pop up. this is something I encountered while testing this fixture.
https://pytest-services.readthedocs.io/en/2.2.1/#infrastructure-fixtures

Reply:
That's only when using the `worker_id` fixture provided by pytest-services. Using the methods provided by xdist itself, this will only be 'master'.

Reply:
I'm curious as to if that is true or not since the fixture version has one line of code `return get_xdist_worker_id(request)`